From e4a3f0317118b8aa654de3d531900dc6373188a7 Mon Sep 17 00:00:00 2001
From: Samin Guo <samin.guo@starfivetech.com>
Date: Fri, 3 Mar 2023 16:59:23 +0800
Subject: [PATCH 28/33] dt-bindings: net: starfive,jh7110-dwmac: Add
 starfive,syscon

A phandle to syscon with two arguments that configure phy mode.

Signed-off-by: Samin Guo <samin.guo@starfivetech.com>
---
 .../bindings/net/starfive,jh7110-dwmac.yaml         | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/Documentation/devicetree/bindings/net/starfive,jh7110-dwmac.yaml b/Documentation/devicetree/bindings/net/starfive,jh7110-dwmac.yaml
index ca49f08d50dd..79ae635db0a5 100644
--- a/Documentation/devicetree/bindings/net/starfive,jh7110-dwmac.yaml
+++ b/Documentation/devicetree/bindings/net/starfive,jh7110-dwmac.yaml
@@ -58,6 +58,18 @@ properties:
       Tx clock is provided by external rgmii clock.
     type: boolean
 
+  starfive,syscon:
+    $ref: /schemas/types.yaml#/definitions/phandle-array
+    items:
+      - items:
+          - description: phandle to syscon that configures phy mode
+          - description: Offset of phy mode selection
+          - description: Mask of phy mode selection
+    description:
+      A phandle to syscon with two arguments that configure phy mode.
+      The argument one is the offset of phy mode selection, the
+      argument two is the mask of phy mode selection.
+
 allOf:
   - $ref: snps,dwmac.yaml#
 
@@ -96,6 +108,7 @@ examples:
         snps,en-tx-lpi-clockgating;
         snps,txpbl = <16>;
         snps,rxpbl = <16>;
+        starfive,syscon = <&aon_syscon 0xc 0x1c0000>;
         phy-handle = <&phy0>;
 
         mdio {
-- 
2.39.1

